packages/vitest/src/reporters/console.ts

```diff
@@ -77,14 +77,26 @@ export class ConsoleReporter implements Reporter {
       }
     }
 
+    const errorMap = new Map<string | undefined, {error: unknown; tests: Test[]} >()
+
     if (failedTests.length) {
       this.ctx.error(c.red(divider(c.bold(c.inverse(` Failed Tests ${failedTests.length} `)))))
       this.ctx.error()
       for (const test of failedTests) {
-        this.ctx.error(`${c.red(c.bold(c.inverse(' FAIL ')))} ${getFullName(test)}`)
-        await printError(test.result?.error, this.ctx)
+        const error = test.result?.error as Error
+        if (errorMap.get(error.stack))
```

Comment:
Should use `.stackStr` instead, as `.stack` might not exist when serialized by worker communication. Also need to do the null and type check, as `test.result.error` might be undefined, or even a string. In that case, we shouldn't merge them.

So I guess Map is not ideal here, I would prefer a nested array and use `.find` to do the match. Might be a bit less efficient but it gives more reliable results, and perf doesn't matter much in this case.

Reply:
sure, i'll refactor again soon